### PR TITLE
[DOCS-7217/7054] Observability Pipelines 2.0 and legacy docs

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

- Moves OP 1.0 docs to "legacy" folder and hides the docs
- Adds OP 2.0 docs (see https://github.com/DataDog/documentation/pull/22171)

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/observability_pipelines/legacy/_index.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+private: true
+cascade:
+    private: true
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">Observability Pipelines is not available on the US1-FED Datadog site.</div>
+{{< /site-region >}}
+
+<div class="alert alert-danger">If you upgrade your Observability Pipelines Workers 1.8 or earlier to 2.0 or later, your existing pipelines will break. Do not upgrade your Workers if you want to continue using the legacy Observability Pipelines. If you want to use Worker 2.0 or later, you must migrate your Worker 1.8 or earlier pipelines to Worker 2.x.</div>
```
Comment (estherk15) [inline content/en/observability_pipelines/legacy/_index.md:28]: I'm guessing that `migrating` and `upgrading` are two different processes. Is there more information on how to migrate to the new worker vs. upgrading?

File: content/en/observability_pipelines/legacy/_index.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+{{< whatsnext desc="Reference: Datadog Processing Language " >}}
+    {{< nextlink href="observability_pipelines/legacy/reference/processing_language/" >}}Overview{{< /nextlink >}}
+    {{< nextlink href="observability_pipelines/legacy/reference/processing_language/functions/" >}}Functions{{< /nextlink >}}
+    {{< nextlink href="observability_pipelines/legacy/reference/processing_language/errors/" >}}Errors{{< /nextlink >}}
+{{< /whatsnext >}}
+
+{{< whatsnext desc="Guides " >}}
+    {{< nextlink href="observability_pipelines/legacy/guide/control_log_volume_and_size/" >}}Control Log Volume and Size{{< /nextlink >}}
+    {{< nextlink href="observability_pipelines/legacy/guide/ingest_aws_s3_logs_with_the_observability_pipelines_worker/" >}}Ingest Amazon S3 Logs with the Observability Pipelines Worker{{< /nextlink >}}
+    {{< nextlink href="observability_pipelines/legacy/guide/route_logs_in_datadog_rehydratable_format_to_Amazon_S3" >}}Route Logs in Datadog-Rehydratable Format to Amazon S3{{< /nextlink >}}
```
Comment (estherk15) [inline content/en/observability_pipelines/legacy/_index.md:68]: This 404s for me, didn't know links were case sensitive
```suggestion
    {{< nextlink href="observability_pipelines/legacy/guide/route_logs_in_datadog_rehydratable_format_to_amazon_s3" >}}Route Logs in Datadog-Rehydratable Format to Amazon S3{{< /nextlink >}}
```

**Review by estherk15 — APPROVED** [review]
Found a few pages without an alias to verify!

File: content/en/observability_pipelines/legacy/monitoring.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Monitoring
+title: (LEGACY) Monitoring
 kind: documentation
 disable_toc: false
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/observability_pipelines/legacy/monitoring.md:4]: This doesn't need an alias. We are are using this for the new OP docs.

File: content/en/observability_pipelines/legacy/guide/_index.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+title: (LEGACY) Observability Pipelines Guides
+kind: guide
+private: true
+disable_toc: true
+aliases:
+    - /integrations/observability_pipelines/guide/
+    - /observability_pipelines/guide/custom-metrics-governance
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/observability_pipelines/legacy/guide/_index.md:8]: That doc was removed a couple of months back.

**Pull request merged**